### PR TITLE
New component: `o-text-introduction`

### LINKED ISSUE #1783: Add Text introduction component to the DS
This issue will track the work of adding the Text introduction component to the CFPB Design System. On the SBL team we're working on adding the Text introduction as a component to the React component library. 

### Text introduction specs
- 45px space above the component (this should come from the parent container)
  - Component top-margin: 0px
- 15px of space between all parts at desktop and mobile 
  - Desktop: H1, Lead paragraph, paragraph, CTA/list link
  - Mobile: H2, Lead paragraph, paragraph, CTA/list link
- Lead paragraph: change to top-margin: 0px and bottom-margin: 15px 
  - The update to the Lead paragraph component is being tracked here: https://github.com/cfpb/design-system/issues/1782

| Existing spacing | Updated spacing |  
|------------| ------------|
|Before|After|

@sonnakim for awareness

### PR DESCRIPTION
**Pull request by meissadia**
Closes #1783

[Deployed preview](https://deploy-preview-1781--cfpb-design-system.netlify.app/design-system/patterns/text-introductions)

## Additions

- Creates `o-text-introduction`
- New graphic of Text Introduction spacing specs

## Removals

- Old example screenshot

## Changes

- Updates Text Introduction docs page with code example and removes outdated references

## Screenshots
|Description|Screenshot|
|---|---|
|Rendered example|Screenshot 2023-10-18 at 4 39 28 PM|
|Code|Screenshot 2023-10-18 at 4 39 35 PM|
|Specs|Screenshot 2023-10-18 at 4 39 40 PM|
|Mobile view|Screenshot 2023-10-18 at 4 45 16 PM|

### CODE REVIEW

File: docs/pages/introductions.md [hunk 1]
```diff
@@ -4,7 +4,28 @@ section: patterns
 status: Released
 variation_groups:
   - variations:
-      - variation_code_snippet: ''
+      - variation_code_snippet: >-
+          <div classname="o-text-introduction">
```
Comment (anselmbradford) [inline docs/pages/introductions.md:8]: You have `classname` here instead of `class`

File: docs/pages/introductions.md [hunk 2]
```diff
@@ -4,7 +4,28 @@ section: patterns
...
+            <h1>Text introduction</h1>
+            <p classname="lead-paragraph">
```
Comment (anselmbradford) [inline docs/pages/introductions.md:10]: As above, should be `class`, not `classname`

**Review by natalia-fitzgerald — APPROVED** [review]
This is approved. For the live code example can we remove the extra link and include only one Call-to-action link?

**Comment by meissadia** [comment]
> This is approved. For the live code example can we remove the extra link and include only one Call-to-action link?

I've not updated the screenshots but have removed the second CTA link.

**Pull request merged**